src/point.rs

```diff
@@ -185,6 +186,30 @@ impl Point {
     pub fn is_nan(self) -> bool {
         self.x.is_nan() || self.y.is_nan()
     }
+
+    /// The topmost [`Point`]
+    #[inline]
+    pub fn topmost(&self, other: &Self) -> Ordering {
+        self.y.total_cmp(&other.y)
+    }
+
+    /// The bottommost [`Point`]
+    #[inline]
+    pub fn bottommost(&self, other: &Self) -> Ordering {
+        other.y.total_cmp(&self.y)
+    }
+
+    /// The rightmost [`Point`]
+    #[inline]
+    pub fn rightmost(&self, other: &Self) -> Ordering {
+        self.x.total_cmp(&other.x)
+    }
+
+    /// The leftmost [`Point`]
+    #[inline]
+    pub fn leftmost(&self, other: &Self) -> Ordering {
+        other.x.total_cmp(&self.x)
+    }
```

Comment:
Is there a more standard way for this type of comparison other than just implementing a function? Maybe we should introduce a trait that size and vec2 can also benefit from?